### ISSUE TITLE
Old torrents keep appearing and cannot be permanently deleted

### ISSUE THREAD

**Issue by jlholt** [transmission/transmission#5043]
### What is the issue?

After upgrading from 2.94 to 4.0.1, I opened the app and there appeared 4 very old torrents that I had long ago removed from the list of torrents.

So I removed them again and they disappear from the list. Or so I thought.

I then quit the app and restarted it. They reappeared.

What diagnostics do you need to investigate this issue?

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.1

**Comment by jlholt** [comment]
I should also mention that I had one torrent in the list before the upgrade that was no longer present after the upgrade.

**Comment by ckerr** [comment]
@jlholt, I have a _theory_ on what's happening and would like you to test something for me to test the theory:

Open up the inspector for one of the torrents that keeps coming back and take note of its infohash string. Then in your home directory, take a look at `Library/Application Support/Transmission/torrents`, see how many files have the first part of that infohash.

Some files will literally be `${infohash}.torrent`

Some older files are `${name}.${first_16_characters_of_infohash}.torrent`

Do you have zero, one, two (or somehow more??) of these files? And for any that you have, what are their timestamps and file permissions (e.g. are they somehow read-only)

**Comment by jlholt** [comment]
Before you replied, I added two torrents and those are of the form
${infohash}.torrent and the data presented in the inspector match the
filenames in that folder. I added these two after removing the 4 "bad"
boomerangs.

Here is a slightly obfuscated view of those files.

$ pwd
/Users/sadsack/Library/Application Support/Transmission/Torrents

mbp 23:42:38
$ ls -l
total 1720
-rw-------@ 1 sadsack  staff     295 Feb 24 23:36
541a48a474b917f77fe5a505243d76ce2774661d.magnet
-rw-------@ 1 sadsack  staff     679 Feb 24 23:36
5d34c2ca518c38d57949e5ebaeddc32359c4c078.magnet
-rw-------@ 1 sadsack  staff     305 Feb 24 23:36
776ceefe5de6773ff97ed5a8391939906898e2c7.magnet
-rw-------@ 1 sadsack  staff   21553 Feb 24 22:10
953e96218b480e1d3a2e1fa543a222ecbed8121b.torrent
-rw-------  1 sadsack  staff   18587 Feb 23 10:02 Blood Orgy of the
She-Devils (1973).5d34c2ca518c38d5.torrent
-rw-------  1 sadsack  staff   25363 Dec 15  2012 Invasion of the Blood
Farmers (1972).776ceefe5de6773f.torrent
-rw-------  1 sadsack  staff   45595 Dec 15  2012 Bloodsucking Pharaohs in
Pittsburgh (1991).541a48a474b917f7.torrent
-rw-------  1 sadsack  staff   39674 Mar 22  2012 Terror at Blood Fart Lake
(2008).af9c9f9c15688295.torrent
-rw-------@ 1 sadsack  staff     244 Feb 24 23:36
af9c9f9c15688295faaafd9f034421ae42171ef7.magnet
-rw-------@ 1 sadsack  staff  699821 Feb 24 22:03
d409fe70b6ca0e6d4ee9e51dda61dd12d3dd502f.torrent
